caom2repo/caom2repo/core.py

```diff
@@ -257,15 +259,15 @@ def put_observation(self, observation):
         """
         assert observation.collection is not None
```

Comment:
I understand that although the cost of checking is minimal since the conditions are very simple. Sometimes it is better to identify the problems earlier instead of letting them trickle down the call stack. BTW, I'm wondering how can we compile the clients with optimization (-O ?) turned on or if we should.